### ISSUE TITLE
Testing Sensor in branch deployment persistently blocked by `dagster._core.errors.DagsterUserCodeUnreachableError`

### ISSUE THREAD

**Issue by BFreeNashville** [dagster-io/dagster#18116]
### Dagster version

1.5.8

### What's the issue?

When attempting to test a sensor via the "Test Sensor" functionality in the UI within a branch deployment, I continue to get the `dagster._core.errors.DagsterUserCodeUnreachableError` error. 

I've tried waiting, redeploying, etc., but the error persists.

Earlier in the day yesterday, I was able to test the sensor and received a valid error from within my code. This new error - while seemingly unrelated - appeared after pushing the fix.
![Screenshot 2023-11-17 at 10 47 27 AM](https://github.com/dagster-io/dagster/assets/33765915/b8a1bd78-51cd-4be8-a519-8df343f95604)

### What did you expect to happen?

The sensor code is located and ran.

### Deployment type

Dagster Cloud

### Deployment details

Having to use a docker deployment due to necessary use of `dagster_cloud_pre_install.sh` and `dagster_cloud_post_install.sh`

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by prha** [comment]
Hi @BFreeNashville ... Are you doing significant work inside of the sensor?  We can do better in our error messages / logging, but cloud users often hit this error message on branch deployments when the sensor evaluation times out (after 45 seconds).

Wondering if you can get a successful evaluation by providing a cursor value / break up the amount of work you're checking for in your evaluation function.

**Comment by BFreeNashville** [comment]
Thanks @prha, looks like that was it. The work in the sensor isn't nothing but I'm surprised it's taking longer than 45 secs. 🤷  Appreciate the help! I did run into a separate issue where I'm getting some unexpected behavior from a sensor with multiple target jobs, but I'll open another ticket. I'll close this out.
